gui/src/components/Notifications.js

```diff
@@ -0,0 +1,66 @@
+import React, { Fragment } from 'react'
+import { Link } from 'react-static'
+import LinkButton from 'components/LinkButton'
+import Flash from './Flash'
+import { receiveSignoutSuccess } from 'actions'
+import { bindActionCreators } from 'redux'
+import { connect } from 'react-redux'
+
+export class Notifications extends React.Component {
+  signOutLocally = () => {
+    this.props.receiveSignoutSuccess()
+  }
+
+  errorPresenter = (err, i) => {
+    if (err.status === 401) {
+      return <p key={i}> {err.detail}
+        <LinkButton onClick={this.signOutLocally}>
+          Sign In Again
+        </LinkButton>
+      </p>
+    }
+    return <p key={i}>{err.detail}</p>
+  }
+
+  render () {
+    const { errors, successes, classes } = this.props
+    return (
+      <Fragment>
+        {errors.length > 0 &&
+          <Flash error className={classes.flash}>
+            {errors.map((err, i) => this.errorPresenter(err, i))}
+          </Flash>
+        }
+        {successes.length > 0 && (
+          <Flash success className={classes.flash}>
+            {successes.map((succ, i) => successPresenter(succ, i))}
+          </Flash>
+        )}
+      </Fragment>
+    )
+  }
+}
+
+const successPresenter = (success, i) => {
```

Comment:
why is errorPresenter within the class while successPreenter is not?

Reply:
Because `errorPresenter` uses `this.signOutLocally`.

Reply:
why can't successPresenter go inside? 🗡

Reply:
It can.